sechub-wrapper-prepare/src/main/java/com/mercedesbenz/sechub/wrapper/prepare/modules/WrapperTool.java

```diff
@@ -0,0 +1,107 @@
+package com.mercedesbenz.sechub.wrapper.prepare.modules;
+
+import static com.mercedesbenz.sechub.commons.pds.PDSDefaultParameterKeyConstants.PARAM_KEY_PDS_CONFIG_PRODUCT_TIMEOUT_MINUTES;
+import static com.mercedesbenz.sechub.wrapper.prepare.cli.PrepareWrapperKeyConstants.KEY_PDS_PREPARE_PROCESS_TIMEOUT_SECONDS;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import javax.crypto.SealedObject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.mercedesbenz.sechub.commons.core.security.CryptoAccess;
+import com.mercedesbenz.sechub.commons.pds.PDSDefaultParameterValueConstants;
+import com.mercedesbenz.sechub.commons.pds.PDSProcessAdapterFactory;
+import com.mercedesbenz.sechub.commons.pds.ProcessAdapter;
+
+@Component
+abstract class WrapperTool {
+
+    private static final Logger LOG = LoggerFactory.getLogger(WrapperTool.class);
+    private static final int defaultMinutesToWaitForProduct = PDSDefaultParameterValueConstants.DEFAULT_MINUTES_TO_WAIT_FOR_PRODUCT;
+
+    @Value("${" + PARAM_KEY_PDS_CONFIG_PRODUCT_TIMEOUT_MINUTES + ":" + defaultMinutesToWaitForProduct + "}")
+    private int pdsProductTimeoutMinutes;
+
+    @Value("${" + KEY_PDS_PREPARE_PROCESS_TIMEOUT_SECONDS + ":-1}")
+    private int pdsPrepareProcessTimeoutSeconds;
+
+    ProcessAdapter process;
+
+    @Autowired
+    PDSProcessAdapterFactory processAdapterFactory;
+
+    @Autowired
+    UserInputValidator userInputValidator;
+
+    private final List<String> forbiddenCharacters = Arrays.asList(">", "<", "!", "?", "*", "'", "\"", ";", "&", "|", "`", "$", "{", "}");
+
+    abstract void cleanUploadDirectory(String uploadDirectory) throws IOException;
+
+    void escapeRepositoryURL(String repositoryURL, List<String> forbiddenCharacters) {
+        if (forbiddenCharacters == null) {
+            forbiddenCharacters = this.forbiddenCharacters;
+        }
+        userInputValidator.validateLocationURL(repositoryURL, forbiddenCharacters);
+    }
+
+    void waitForProcessToFinish() {
```

Comment:
Instead of having a field of the type `ProcessAdapter` inside this class, I would recommend to change this method using it. It is better to read for which process we are waiting and if this method gets called before the `ProcessAdapter` was initialized we get a NullpointerException here.

```suggestion
    void waitForProcessToFinish(ProcessAdapter process) {
```